File: Python/Examples1/example1_1.py
"""
example1_1.py
(c) Copyright 2022-2025, Shogo MURAMATSU, All rights reserved
"""
import sys

# 電源電圧値
SRC_VOLTAGE = 5.0
# 抵抗の数
NUM_REGS = 5

def main(args):
    if len(args) != 6:
        print('エラー：引数の数' + str(len(args)))
        sys.exit(0)

    # 抵抗値 Ri
    r = [ 0.0 for idx in range(NUM_REGS) ]
    for idx in range(NUM_REGS):
        r[idx] = float(args[idx+1])
    # 抵抗R0にかかる電圧V
    e = SRC_VOLTAGE
    r0, r1, r2, r3, r4 = r[0], r[1], r[2], r[3], r[4]
    v0 = (r2*r3-r1*r4)*r0*e / \
        ( r1*r3*(r2+r4) + r2*r4*(r1+r3) + r0*(r1+r3)*(r2+r4) )            

    # 電圧Vの表示
    print('電源電圧 E = ' + str(SRC_VOLTAGE) + ' V')
    for idx in range(NUM_REGS):
        print('抵抗 R' + str(idx) + ' = ' + str(r[idx]) + ' Ω')
    print('電圧 V0 = ' + str(v0) + ' V')

if __name__ == '__main__':
    main(sys.argv)
        
        

File: C/Examples1/example1_1.c
/*
 * example1_1.c
 *
 * プログラミングBI 演習例題1-1
 *
 * Copyright (C) 2007-2025, S. Muramatsu
 *
 */
#include <stdio.h>

int main()
{
	/* 変数宣言 */
	double x; /* 独立変数 */
	double y; /* 従属変数 (関数の近似値) */
	int n; /* 近似次数 */

	/* 途中の計算に必要な変数の宣言 */
	int i; /* 繰り返し変数の宣言 */
	double f; /* 階乗値 */
	double p; /* x の累乗値 */

	/* データの読み込み(倍精度, 整数) */
	scanf("%lf%d",&x,&n);

	/* 近似値の初期化 */
	y = 1.0;
	f = 1.0;
	p = 1.0;
	for (i=1; i<=n; i++) {
		f = f * (double)i; /* f *= (double)i; とも書ける */
		p = p * x; /* p *= x; とも書ける */
		y = y + p / f; /* y += p / f; とも書ける */
	}

	/* e^x の近似値の表示 */
	printf("マクローリン展開による次数 %d までの e^(%g) の近似値\n"
		"    =%21.16g\n", n, x, y);
	
	return 0;
}
